src/OpenTelemetry.AutoInstrumentation/Diagnostics/SdkSelfDiagnosticsEventListener.cs

```diff
@@ -88,24 +91,32 @@ protected override void OnEventSourceCreated(EventSource eventSource)
     /// <param name="eventData">Data of the EventSource event.</param>
     protected override void OnEventWritten(EventWrittenEventArgs eventData)
     {
-        var payloadArray = new object[eventData.Payload.Count];
-        eventData.Payload.CopyTo(payloadArray, 0);
+        object[] payloadArray;
+        if (eventData.Payload != null)
+        {
+            payloadArray = new object[eventData.Payload.Count];
+            eventData.Payload.CopyTo(payloadArray, 0);
+        }
+        else
+        {
+            payloadArray = Array.Empty<object>();
+        }
 
         switch (eventData.Level)
         {
             case EventLevel.Critical:
             case EventLevel.Error:
-                log.Error("EventSource={0}, Message={1}", eventData.EventSource.Name, string.Format(eventData.Message, payloadArray));
+                log.Error("EventSource={0}, Message={1}", eventData.EventSource.Name, string.Format(eventData.Message ?? string.Empty, payloadArray));
```

Comment:
for refactoring > `string.Format(eventData.Message ?? string.Empty, payloadArray)` could be extracted to local var, too many repetitions.

Reply:
16ce6223c259f140db177d5c08e2ca521f8366d2